src/substrait/textplan/README.md

```diff
@@ -0,0 +1,208 @@
+
+```
+"0123456789abcde"_binary
+"1234"_fixedbinary<2>
+"00-ABCD-1234"_uuid
+```
+
+### Time Literals
+
+There are two kinds of time literals:  literals that represent particular points in time and literals that represent
+durations.  The timestamps represent dates from the beginning of the year 1000 to the end of the year 9999.  The 
+interval_year duration has its precision in months and the interval_day duration has its precision in microseconds.
+
+To make the structured data easier to read the pseudo types ```year```, ```month```, ```day```, ```second```, and
+```microsecond``` (plural or singular) are suggested for readability.
+
+#### Example Time Literals
+
+```
+"1000-01-01 00:00:00.000000"_timestamp
+"1000-01-01 00:00:00.000000 UTC"_timestamp_tz
+"2020-12-20"_date
+"13:21"_time
+"13:21:12.012345"_time
+
+{5_years, 1_month}_interval_year
+{5_year, 1_months}_interval_year
+{4_days, 1_second, 13_microseconds}_interval_day
+```
+
+### Structured Literals
+
+Structure literals use the brace syntax.  Types are not necessary within the value section as the types should be 
+homogeneous if the structure is well-formed. 
+
+|                                                           | Description                                   |
+|-----------------------------------------------------------|-----------------------------------------------|
+| {}_list\<string>                                          | empty list of string                          |
+| {}_list\<string?>                                         | empty list of optional strings                |
+| {null, "a", "b"}_list\<string?>                           | list of optional strings                      |
+| {}_map\<int, string>                                      | empty map with integer keys and string values |
+| {42->"life", 32->"everything"}_map\<int, string>          | map with integer keys and string values       |
+| {"a", {"b", "c"}_struct\<string, struct\<string, string>> | structure with nested substructure            |
+
+#### Example Structured Literals
+```
+{"a", "b", "c"}_list<string>
+
+```
+
+### Boolean Literals
+
+The values for boolean literals are distinct from all other literals so appending the type is optional.
+
+#### Example Boolean Literals
+```
+true
+false
+true_bool
+false_bool
+```
+
+### Null
+
+Null also has a distinct form.  Types are still necessary to inform which function is to be used to process the literal.
+All null types must include an optional value (so that the null can be provided) and thus normally would end with a 
+question mark to show that it is optional.  For brevity the question mark may be eliminated.
+
+Nulls can represent either an unknown value or the lack of a value depending on the semantics of the operation.  (A sum
+function using null as unknown would result in unknown, but a sum operation treating null as not present would result in
+the total of the non-null elements.)
+
+#### Example Null Literals
+```
+null_i32
```

Comment:
Can you add a comment mentioning that the type is implicitly `i32?`?

Reply:
done